src/ray/rpc/client_call.h

```diff
@@ -123,6 +129,11 @@ class ClientCallImpl : public ClientCall {
   /// The stats handle tracking this RPC.
   std::shared_ptr<StatsHandle> stats_handle_;
 
+  /// Whether to record grpc_client_req_failures for this.
+  /// Recorded for everything except gcs client and clients on workers.
+  /// All rpc's to gcs are always retried anyways.
+  bool record_stats_;
```

Comment:
if it's only controlling the single metric, would rename accordingly:
```suggestion
  bool record_num_request_failures_;
```

can be made abstract if/when we extend it

Reply:
this is also an abstraction leak:

>   /// Recorded for everything except gcs client and clients on workers.

it'll very likely become out-of-date when the callsites change and then be misleading